Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -121,6 +123,19 @@ - (void)setupWith:(SentryReplayOptions *)replayOptions
     }
 
     _notificationCenter = SentryDependencyContainer.sharedInstance.notificationCenterWrapper;
+    _dateProvider = SentryDependencyContainer.sharedInstance.dateProvider;
+
+    // The asset worker queue is used to work on video and frames data.
+    // Use a relative priority of -1 to make it lower than the default background priority.
+    _replayAssetWorkerQueue = [SentryDispatchQueueWrapper
+        createBackgroundDispatchQueueWithName:"io.sentry.session-replay.asset-worker"
+                             relativePriority:-1];
```

Comment:
`h`: That's so easy to break. If we ever change the priority of the default background priority here, we could end up with the same problem again. And when I change the priority in the `SentryDispatchQueueWrapper` how should I know that I I need to change it here too?

https://github.com/getsentry/sentry-cocoa/blob/8856e5a7cf210e0417952d909d5924e76a7d1620/Sources/Sentry/SentryDispatchQueueWrapper.m#L10-L13

A test should fail if these priorities don't match our requirements here. If you need ideas for testing this properly, please let me know.

Reply:
I would highly appreciate an example how you would test the priorities of queues.

Reply:
Using a factory method that I can swap out in the tests or something like that. Then I can assert that I call the factory with the correct parameters.

Reply:
Thinking about this again, I am starting to question if another factory makes sense, because the `createBackgroundDispatchQueueWithName` already is a factory.
But I do see your point of this breaking easily and that we do not have tests to validate the behavior.

Therefore I am going to implement this:
- I'll add a new protocol `SentryDispatchQueueProviderProtocol` and an instance of `SentryDispatchQueueProvider`
- The provider will then use the `SentryDispatchQueueWrapper` under the hood to create dispatch queues
- This will make it replaceable in tests so we can assert that the queues are created with the correct arguments.
- The provider will live in the container, as there is no possibility to inject a dependency here, as we are already in the top-level integration

Reply:
Nevermind, I just noticed that the `SentryDispatchQueueWrapper` already is used as an instance to create queues, so I'll just change the new method to be non-static

Reply:
I just found `SentryDispatchFactory`, will extend this

Reply:
alright, added the protocol and tests for the queues.
I actually did not mock the factory in the unit tests, but instead just checked the priorities of the queues underneath. Still believe there is value in adopting the factory pattern.